README.md

```diff
@@ -1,47 +1,178 @@
+3. Generate the dotnet, Go, Node, and Python SDKs and place them in the `./sdk` folder
+4. Install the provider on your machine.
+
+#### Test against the example
+   
+```bash
 $ cd examples/simple
 $ yarn link @pulumi/xyz
 $ yarn install
 $ pulumi stack init test
 $ pulumi up
 ```
 
+Now that you have completed all of the above steps, you have a working provider that generates a random string for you.
+
+#### A brief repository overview
+
+You now have:
+
+1. A `provider/` folder containing the building and implementation logic
+   1. `cmd/`
+      1. `pulumi-gen-xyz/` - generates language SDKs from the schema
+      2. `pulumi-resource-xyz/` - holds the package schema, injects the package version, and starts the gRPC server
+   2. `pkg`
+      1. `provider` - holds the gRPC methods (and for now, the sample implementation logic) required by the Pulumi engine
+      2. `version` - semver package to be consumed by build processes
+2. `deployment-templates` - a set of files to help you around deployment and publication
+3. `sdk` - holds the generated code libraries created by `pulumi-gen-xyz/main.go`
+4. `examples` a folder of Pulumi programs to try locally and/or use in CI.
+5. A `Makefile` and this `README`.
+
+### Writing the schema
+
+The [JSON schema file](https://www.pulumi.com/docs/guides/pulumi-packages/schema) is used by `pulumi-gen-xyz` to create language-specific SDKs. 
+It is, therefore, a central requirement for any resource provider.
+Provider schemas can be machine-generated from an API specification, or handwritten.
+This repository provides the [xyz example schema](./provider/cmd/pulumi-resource-xyz/schema.json) to get you started.
+[The Azure Native Provider schema](https://raw.githubusercontent.com/pulumi/pulumi-azure-native/master/provider/cmd/pulumi-resource-azure-native/schema.json) provides a much larger example.
+Refer to the [package schema documentation](https://www.pulumi.com/docs/guides/pulumi-packages/schema/#pulumi-package-schema) for additional details.
+
+### Implementing the gRPC methods
+
+Once you have a schema that describes all the resources for your provider, you will need to implement the desired gRPC methods.
+You will find a mostly blank implementation of these in `pkg/provider/provider.go`.
+Note that these methods do not necessarily link 1:1 to the Pulumi CLI commands.
+
+#### Basic Functionality
+
+The struct and creation of the provider are implemented already:
+
+```go
+// provider/pkg/provider.go
+type xyzProvider struct {
+	host    *provider.HostClient
+	name    string
+	version string
+	schema  []byte
+}
+
+func makeProvider(host *provider.HostClient, name, version string, pulumiSchema []byte) (pulumirpc.ResourceProviderServer, error) {
+   // Return the new provider
+   return &xyzProvider{
+   host:    host,
+   name:    name,
+   version: version,
+   schema:  pulumiSchema,
+   }, nil
+}
+```
+
+You need to provide the following methods _for each resource_:
+
+1. Check
```

Comment:
is it for each resource? I can't see 100s of sets of these for pulumi-aws-native